src/components/Button/index.tsx

```diff
@@ -294,6 +307,80 @@ export function ButtonWithLink({ link, text, style }: { link: string; text: stri
   )
 }
 
+export function ButtonExternalLink({
+  link,
+  children,
+  disabled = false,
+}: {
+  link: string
+  disabled?: boolean
+  children: ReactNode
+}) {
+  const theme = useContext(ThemeContext)
+  if (disabled)
+    return (
+      <ButtonPurpleDim disabled={disabled}>
+        {children}
+        <Box ml={2}>
+          <ArrowUpRight size="14px" color={theme.purple2} />
+        </Box>
+      </ButtonPurpleDim>
+    )
+
+  return (
+    <ButtonPurpleDim disabled={disabled} as={ExternalLink} href={link}>
```

Comment:
Can we do something like this here?

```suggestion

  return (
    <ButtonPurpleDim disabled={disabled} as={!disabled && ExternalLink} href={!disabled && link}>
```

Reply:
Yup, I've wanted to simplify this but forgot. Thanks.